src/UnitConversions.hs

```diff
@@ -54,3 +60,42 @@ convertIngredientToImperial ingr =
     Quart   -> ingr
     Gallon  -> ingr
     Other _ -> ingr
+
+
+convertTemperatureToMetric = convertTemperature C
+convertTemperatureToImperial = convertTemperature F
+
+convertTemperature :: TempUnit -> String -> String
+convertTemperature u s = unpack $ foldl replaceTemperature (pack s) (fmap packText $ fmap convertReplacement (findReplacements s))
+  where packText (s1, s2) = (pack s1, pack s2)
+        replaceTemperature text (old, new) = replace old new text
+        convertReplacement = fmap $ show . toTempUnit u
+
+findReplacements :: String -> [(String, Temperature)]
+findReplacements = map parseRegexResult . findTemperatures
+  where parseRegexResult l = (l!!0, Temperature (read (l!!1)) (parseTempUnit (l!!2)))
```

Comment:
Can we avoid using the partial function `!!` here? Perhaps the result could be in a `Maybe`?

Reply:
replaced `read` by `readMaybe` from Text.Read: http://hackage.haskell.org/package/base-4.12.0.0/docs/Text-Read.html#v:readMaybe

Replaced `!!` with a new `at` method. I couldn't find an existing method that does this.
I found `Lens.Micro (ix)`, but `Lens.Micro` is only a dependency of the herms executable and not the library, and I am not sure if it is ok to add `Lens.Micro` just for that.

Reply:
@MoritzR I think it'd be fine to add `microlens` as a dependency to get `ix`.

Reply:
Pattern matching actually removed the need for this method!